### ISSUE TITLE
Memory Leaks in base project

### ISSUE THREAD

**Issue by sean-perkins** [NativeScript/NativeScript#4700]
### Did you verify this is a real problem by searching [Stack Overflow]
Yes

### Tell us about the problem
I am in the process of debugging and optimizing our own application to reduce memory consumption and consistent app crashes. The application is in NS + Angular, and when attaching to xCode debugger, I found substantial traces and warnings of memory leaks. 

I narrowed the scope of the problem by creating a new instance of a vanilla NativeScript project and also attaching it to the debugger. What I found was that xCode is warning about memory leaks on the base project as well.

Trying to optimize our application is difficult when we don't have a safety point to refer to for best practices and consistency.

### Which platform(s) does your issue occur on?
iOS (did not confirm on Android)

### Please provide the following version numbers that your issue occurs with:
- CLI: 3.1.2
- Cross-platform modules: 3.1.1
- Runtime(s): 3.1.0
- Plugin(s): N/A

### Please tell us how to recreate the issue in as much detail as possible.

1. Create a new project `tns create DemoMemoryLeak`
2. Run the application
3. Attach xCode debugger to the application
4. View the memory graph

![screen shot 2017-08-15 at 1 29 25 pm](https://user-images.githubusercontent.com/13732623/29327750-2d2e4b62-81be-11e7-8d08-31d30ee26deb.png)

5. After clicking all the "tap" attempts in the hello world demo, the number jumps from 20 to 27 discovered leak types.

Note: In our flushed out application, we had over 300 identified problems for memory leaks. It's safe to assume that whatever the "problem" is, it's compounded as you grow your application's complexity. 

Attachment of demo app with instruments debugging:
![screen shot 2017-08-15 at 1 53 43 pm](https://user-images.githubusercontent.com/13732623/29328623-2e0a2d3c-81c1-11e7-8b7e-0f4472e5a409.png)

To compare this issue against another app, I ran the same test with an ionic application:

![screen shot 2017-08-15 at 2 07 22 pm](https://user-images.githubusercontent.com/13732623/29329200-14a202aa-81c3-11e7-89a6-8c200474bee2.png)

Let's talk solutions 👍

**Comment by sean-perkins** [comment]
Additional debug information:

- Downgrading to 2.5.0 for `tns-ios` had no measurable impact (per issue #4490)
- The initial memory leak occurs with no interaction on the interface. It's approximately 5.4 seconds after the runtime (consistently). 

Instruments trace: https://drive.google.com/drive/folders/0B9zJUhB5psVdSmZ0ajZienZraTA?usp=sharing

**Comment by NickIliev** [comment]
Related to [https://github.com/NativeScript/NativeScript/issues/4490](https://github.com/NativeScript/NativeScript/issues/4490) (closing as duplicate)

**Comment by jorgerb79** [comment]
Hi, I've created a new project with: "tns create TestApp --ng", and after I installed the app in a real device and opened Instruments->Leaks, the issue persists. The memory is increased each time I choose a player and go back to the list screen, and is not release. 

Also I have a bigger project that uses Google Maps, and the memory is increased dramatically until it fills up the entire device memory.

Note: The output of "tns --version" is 3.3.1
